File: go-sdk/pkg/models/shared/filerequest.go
package shared

import (
	"time"
)

type FileRequestFolderBaseTypeEnum string

const (
	FileRequestFolderBaseTypeEnumFolder FileRequestFolderBaseTypeEnum = "folder"
)

// FileRequestFolderBase
// The folder that this file request is associated with.
//
// Files submitted through the file request form will be
// uploaded to this folder.
type FileRequestFolderBase struct {
	Etag       *string                       `json:"etag,omitempty"`
	ID         string                        `json:"id"`
	Name       *string                       `json:"name,omitempty"`
	SequenceID map[string]interface{}        `json:"sequence_id,omitempty"`
	Type       FileRequestFolderBaseTypeEnum `json:"type"`
}

type FileRequestStatusEnum string

const (
	FileRequestStatusEnumActive   FileRequestStatusEnum = "active"
	FileRequestStatusEnumInactive FileRequestStatusEnum = "inactive"
)

type FileRequestTypeEnum string

const (
	FileRequestTypeEnumFileRequest FileRequestTypeEnum = "file_request"
)

type FileRequestUserBaseTypeEnum string

const (
	FileRequestUserBaseTypeEnumUser FileRequestUserBaseTypeEnum = "user"
)

// FileRequestUserBase
// The user who last modified this file request.
type FileRequestUserBase struct {
	ID    *string                     `json:"id,omitempty"`
	Login string                      `json:"login"`
	Name  string                      `json:"name"`
	Type  FileRequestUserBaseTypeEnum `json:"type"`
}

// FileRequest
// A standard representation of a file request, as returned
// from any file request API endpoints by default.
type FileRequest struct {
	CreatedAt             time.Time              `json:"created_at"`
	CreatedBy             *UserMini              `json:"created_by,omitempty"`
	Description           *string                `json:"description,omitempty"`
	Etag                  *string                `json:"etag,omitempty"`
	ExpiresAt             *time.Time             `json:"expires_at,omitempty"`
	Folder                FileRequestFolderBase  `json:"folder"`
	ID                    *string                `json:"id,omitempty"`
	IsDescriptionRequired *bool                  `json:"is_description_required,omitempty"`
	IsEmailRequired       *bool                  `json:"is_email_required,omitempty"`
	Status                *FileRequestStatusEnum `json:"status,omitempty"`
	Title                 *string                `json:"title,omitempty"`
	Type                  *FileRequestTypeEnum   `json:"type,omitempty"`
	UpdatedAt             time.Time              `json:"updated_at"`
	UpdatedBy             *FileRequestUserBase   `json:"updated_by,omitempty"`
	URL                   *string                `json:"url,omitempty"`
}

File: sdks/java/lib/src/main/java/models/shared/FileRequest.java
package .models.shared;

import java.time.OffsetDateTime;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import .utils.DateTimeSerializer;
import .utils.DateTimeDeserializer;

/**
 * FileRequest
 * A standard representation of a file request, as returned
 * from any file request API endpoints by default.
**/
public class FileRequest {
    @JsonSerialize(using = DateTimeSerializer.class)
    @JsonDeserialize(using = DateTimeDeserializer.class)
    @JsonProperty("created_at")
    public OffsetDateTime createdAt;
    public FileRequest withCreatedAt(OffsetDateTime createdAt) {
        this.createdAt = createdAt;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("created_by")
    public UserMini createdBy;
    public FileRequest withCreatedBy(UserMini createdBy) {
        this.createdBy = createdBy;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("description")
    public String description;
    public FileRequest withDescription(String description) {
        this.description = description;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("etag")
    public String etag;
    public FileRequest withEtag(String etag) {
        this.etag = etag;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("expires_at")
    public OffsetDateTime expiresAt;
    public FileRequest withExpiresAt(OffsetDateTime expiresAt) {
        this.expiresAt = expiresAt;
        return this;
    }
    @JsonProperty("folder")
    public FileRequestFolderBase folder;
    public FileRequest withFolder(FileRequestFolderBase folder) {
        this.folder = folder;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("id")
    public String id;
    public FileRequest withId(String id) {
        this.id = id;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("is_description_required")
    public Boolean isDescriptionRequired;
    public FileRequest withIsDescriptionRequired(Boolean isDescriptionRequired) {
        this.isDescriptionRequired = isDescriptionRequired;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("is_email_required")
    public Boolean isEmailRequired;
    public FileRequest withIsEmailRequired(Boolean isEmailRequired) {
        this.isEmailRequired = isEmailRequired;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("status")
    public FileRequestStatusEnum status;
    public FileRequest withStatus(FileRequestStatusEnum status) {
        this.status = status;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("title")
    public String title;
    public FileRequest withTitle(String title) {
        this.title = title;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("type")
    public FileRequestTypeEnum type;
    public FileRequest withType(FileRequestTypeEnum type) {
        this.type = type;
        return this;
    }
    @JsonSerialize(using = DateTimeSerializer.class)
    @JsonDeserialize(using = DateTimeDeserializer.class)
    @JsonProperty("updated_at")
    public OffsetDateTime updatedAt;
    public FileRequest withUpdatedAt(OffsetDateTime updatedAt) {
        this.updatedAt = updatedAt;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("updated_by")
    public FileRequestUserBase updatedBy;
    public FileRequest withUpdatedBy(FileRequestUserBase updatedBy) {
        this.updatedBy = updatedBy;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("url")
    public String url;
    public FileRequest withUrl(String url) {
        this.url = url;
        return this;
    }
}